torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -588,15 +589,15 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams& params) {
 
   // Figure out which inputs to cache for unrolling or vectorization
   for (auto inp : input_tvs) {
-    if (inp->uses().empty()) {
+    if (inp->uses().empty() || inp->isFusionOutput()) {
       continue;
     }
     cached_inputs.emplace_back(inp->cache_after());
   }
 
   // Figure out which outputs to cache for unrolling or vectorization
   for (auto out : output_tvs) {
-    if (out->definition() == nullptr) {
+    if (out->definition() == nullptr || out->isFusionInput()) {
```

Comment:
Can a fusion input have a definition? Does that happen with a segmented fusion?

Reply:
User could define operations to make a TV and then set that TV as an input. It'd be silly but definitely possible. Segmentation is done by copying the fusion and moving the inputs/outputs so yes, I believe it can.

Reply:
No, I'm wrong...
```.cpp
  // Returns the Expr that this value is an output of, returns nullptr if none
  // was found
  Expr* definition() const {
    if (is_fusion_input_) {
      return nullptr;
    }
    return definition_;
  }
```

Reply:
But then, are those expressions preceding a fusion input considered part of the fusion? For example, `IterVisitor::traveres(Fusion*)` would traverse those pre-input expressions as well as it doesn't stop at fusion inputs, which may or may not make sense depending on use cases.

Reply:
Ah, commented before I saw your last comment.